### PR TITLE
[Azure Search] Introduce last of the breaking changes for Azure search major version revision

### PR DESCRIPTION
**Pull request by arv100kri**
This commit introduces the breaking changes from the [swagger pull request](https://github.com/Azure/azure-rest-api-specs/pull/6934) into the .NET SDK.

The breaking changes that'll go out as part of the next major version release are:

- Skills can now optionally take in a `name` property
- WebApiSkill has an optional `degreeOfParallelism` property, which can allow customers to specify how many API calls (if non-null, non-zero) can be made in parallel to their endpoint
- Add more information to indexer execution's errors and warnings

### CODE REVIEW

**Comment by arv100kri** [comment]
NOTE: This is a draft PR, because the official swagger PR hasn't yet been merged.
Once that's merged, I'll regenerate from the swagger in the master branch of the official repo and add another commit

File: sdk/search/Microsoft.Azure.Search/tests/Tests/IndexerTests.cs [hunk 1]
```diff
@@ -334,13 +334,22 @@ public void CanRunIndexerAndGetIndexerStatus()
 
                 Assert.Equal("1", oldestResult.Errors[0].Key);
                 Assert.Equal("Key field contains unsafe characters", oldestResult.Errors[0].ErrorMessage);
+                Assert.Equal("errorSourceName", oldestResult.Errors[0].Name);
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/IndexerTests.cs]: Why do the expected values look like property names? This is confusing. Where are these values coming from?

File: sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs [hunk 2]
```diff
@@ -1113,7 +1113,8 @@ private static Skillset CreateTestSkillsetWebApiSkill(bool includeHeader = true)
                     description: "A simple web api skill",
                     context: RootPathString)
             {
-                HttpMethod = "POST"
+                HttpMethod = "POST",
+                DegreeOfParallelism = 7
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs]: Where's the test that looks for this 7 in a deserialized Web API skill?

**Review by bitsforthought — CHANGES_REQUESTED** [review]

File: sdk/search/Microsoft.Azure.Search/tests/Tests/IndexerTests.cs (hunk 1 above)
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/IndexerTests.cs]: This is what we mocked from the API - I'll update the mock to be better and then in a later commit fix this

File: sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs (hunk 2 above)
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs]: This is asserted in `CreateAndValidateSkillset` where we check that the response of the create request (response body of a 201) matches what we passed in as a request

**Review by bitsforthought — APPROVED** [review]

**Comment by arv100kri** [comment]
> NOTE: This is a draft PR, because the official swagger PR hasn't yet been merged.
> Once that's merged, I'll regenerate from the swagger in the master branch of the official repo and add another commit

The swagger PR has been merged, and this PR is now "official"

**Pull request merged**